### ISSUE TITLE
Headers and query params

### ISSUE THREAD

**Issue by StalkAlex** [saintsystems/odata-client-php#23]
How can I pass them? Could you show an example please?

**Comment by calebtr** [comment]
If it is helpful, I was able to pass headers this way:

    $odataClient = new ODataClient($serviceURL, function($request) {
        $request->headers['APIKEY'] = 'SuperDuperSecretKey';
    });

For query parameters I use `->where()` for oData's `filter`.

**Comment by StalkAlex** [comment]
yes, it could be done like this, but in my case I need to change them every time before new request. So I solved this problem by creating ***ODataRequest*** instance for every request. But this is not ideal solution.

**Comment by anderly** [comment]
@StalkAlex, thanks for opening this.

I think we can achieve this by adding a `headers` or `withHeaders` method onto the QueryBuilder to allow for appending to/modifying the headers collection set on the ODataRequest.

Can you give an example of the headers you're wanting to set? Just curious.

**Comment by StalkAlex** [comment]
@anderly  currently I found way to get around this problem. Here's my way and example of header

```php
            $request = new ODataRequest('POST', $requestUrl, $this->client);
            $request->addHeaders([
                'Content-Type' => 'text/xml'
            ]);
            $request->attachBody($fileContent);
            $response = $request->execute();
```

**Comment by ds620107** [comment]
So how do I add a custom parameter to query? 

For example I need:
https://api.domain.com/v1/Orders?exported=false
